src/app/features/tech-record/components/tech-record-generate-plate/tech-record-generate-plate.component.ts

```diff
@@ -33,11 +33,11 @@ export class GeneratePlateComponent implements OnInit {
     public userService: UserService,
     private technicalRecordService: TechnicalRecordService
   ) {
-    this.emailAddress$ = this.technicalRecordService.viewableTechRecord$.pipe(
+    this.emailAddress$ = this.store.select(selectTechRecord).pipe(
       tap(record => {
-        if (record?.vehicleType !== 'hgv' && record?.vehicleType !== 'trl') this.navigateBack();
+        if (record?.techRecord_vehicleType !== 'hgv' && record?.techRecord_vehicleType !== 'trl') this.navigateBack();
       }),
-      map(record => record?.applicantDetails?.emailAddress)
+      map(record => record?.techRecord_applicantDetails_emailAddress ?? undefined)
```

Comment:
```suggestion
      map(record => record?.techRecord_applicantDetails_emailAddress)
```
not sure the `?? undefined` is needed?